tools/ctl/ctl.c

```diff
@@ -238,13 +299,46 @@ int main(int argc, char *argv[])
 		/* Print the read EQ configuration data with similar syntax
 		 * as the input file format.
 		 */
-		config = (uint32_t *) (user_data + 2);
-		n = user_data[1] / sizeof(uint32_t);
-		for (i = 0; i < n; i++) {
-			if (i == n - 1)
-				fprintf(stdout, "%u\n", config[i]);
-			else
-				fprintf(stdout, "%u,", config[i]);
+
+		if (binary) {
```

Comment:
We wont need the ```if (binary)``` if we convert ASCII to binary early on. That way the code path is identical whether we are sending ASCII or binary data. The would also mean moving the ASCII -> binary (and vice versa) functions out of ```main()``` too.

Reply:
Um, there might be requirement to dump the control values in ASCII CSV format(to compare with the input CSV file, @singalsu might like that), that's why I kept this if (binary) for .get() output also. People might run "./sof-ctl -n 73 -b -s *.txt" to set with CSV input file, and then run either "./sof-ctl -n 73 -b" or "./sof-ctl -n 73" to dump the TLV kcontrol values in binary format or in CSV txt format, just as their like.

@lgirdwood make sense?

Reply:
@keyonjie Can you move this big if-else into separate function?

Reply:
I can do it like this:
```
dump_data() {
    if (binary)
        dump_binary_data(bool raw);
    else
        dump_tlv_data();
}
```
@jajanusz is this what you want?